File: romanToint.php
<?php


class intRoman
{
    // private key digit to Roman dicttionary
    private static $romanDic = [
        1 => 'I',
        5 => 'V',
        10 => 'X',
        50 => 'L',
        100 => 'C',
        500 => 'D',
        1000 => 'M'
    ];


    /**
     * convert int to roman letter
     *
     * @param integer $int
     * @return string roman letter
     */
    public static function intToRoman(int $int): string
    {
        $roman = [];
        $intArr = str_split((string)$int);
        $len = strlen((string)$int);
        foreach ($intArr as $place => $digit) {
            // $roman[$place] = self::convertDigitToRoman($digit, $len - $place);
            $roman[$place] = self::getRomanByDigitAndPlace($digit, $len - $place);
        }
        $romanString =  implode($roman);

        if($int != self::romanToInt($romanString)){
            echo 'Wrong----'.PHP_EOL;
        }

        return $romanString;
    }

    /**
     * repeat the roman letter 
     *
     * @param string $romanDigit
     * @param integer $repeat
     * @return string
     */
    private static function getRepeatRomanDigit(string $romanDigit, int $repeat): string
    {
        $rtn = '';
        for ($i = 0; $i < $repeat; $i++) {
            $rtn .= $romanDigit;
        }
        return $rtn;
    }


    private static function getRomanByDigitAndPlace(int $digit, int $place):string{
        $roman = ''; 
        switch ($digit) {
            case 0:
                $roman = '';
                break;
            case 4:
                $roman = self::$romanDic[str_pad(1,$place,"0")] . self::$romanDic[str_pad(5,$place,"0")];
                break;
            case 9:
                $roman = self::$romanDic[str_pad(1,$place,"0")] . self::$romanDic[str_pad(1,$place+1,"0")];
                break;
            case 1:
            case 5:
                $roman = self::$romanDic[str_pad($digit, $place, "0")];
                break;
            default:
                $roman = 5 - $digit > 0 ? (self::getRepeatRomanDigit( self::$romanDic[str_pad(1,$place,"0")], $digit)) :  self::$romanDic[str_pad(5,$place,"0")] . (self::getRepeatRomanDigit( self::$romanDic[str_pad(1,$place,"0")],  $digit - 5));
                break;
        }
        return $roman;
    }

    /**
     * convert Roman Letter into Int
     *
     * @param string $roman
     * @return integer
     */
    public static function romanToInt(string $roman):int{
        $romanArr = str_split($roman);
        $intArr = array();

        foreach($romanArr as $place=>$romanDigit){
            $intDigit = self::convetRomanDigitToInt($romanDigit);

            if(isset($intArr[$intDigit])){
                $intArr[$intDigit]+=$intDigit;
            }else{
                $intArr[$intDigit]=$intDigit;

            }
        }
        $lastKey = null;
        foreach($intArr as $key=>$value){
            
            if($lastKey){
                if($lastKey<$key){
                    $intArr[$key] = $value - $intArr[$lastKey];
                    unset($intArr[$lastKey]);
                }
            }
            
            $lastKey = $key;
        }

        return array_sum($intArr);
    }

    private static function convetRomanDigitToInt(string $romanDigit):?int{
        return isset(array_flip(self::$romanDic)[$romanDigit])?array_flip(self::$romanDic)[$romanDigit]:null;
    }


}


do{

    $handle = fopen("php://stdin", "r");
    echo "please type in a INT between 1 and 3999 Or Roman Number: ";
    
    $input = strtoupper(trim(fgets($handle)));
    
    if((int)$input == $input){
        echo "Roman For {$input} is ".  intRoman::intToRoman((int)$input) .PHP_EOL;
        
        
    }else{
        
            echo "INT For {$input} is ".  intRoman::romanToInt($input) .PHP_EOL;
        
    }
}
while(true);

File: romanToint.py
from posixpath import split
import string


class intRoman:
    RomanDic = {
        '1' : 'I',
        '5' : 'V',
        '10' : 'X',
        '50' : 'L',
        '100' : 'C',
        '500' : 'D',
        '1000' : 'M'
    }

    def __init__(self, intNum:int) -> None:
        self.intNum = intNum
        pass

    def intToRoman(self)->str:
        # roman = dict()
        roman = []
        intStr = str(self.intNum)
        intDic = self.split(intStr)
        intLength = len(intStr)
        place = 0
        for digit in intDic:
            roman.append(self.getRomanByDigitAndPlace(int(digit),intLength - place))
            place= place+1
        
        return ''.join(roman)

    
    def getRomanByDigitAndPlace(self, digit, place):
        output = str(digit).ljust(place,'0')
        roman = ''
        if(digit ==0):
            roman = '';
        
        elif(digit == 4):
            roman = self.RomanDic['1'.ljust(place,'0')] + self.RomanDic['5'.ljust(place,'0')]
        
        elif(digit == 9):
            roman = self.RomanDic['1'.ljust(place,'0')] + self.RomanDic['1'.ljust(place+1,'0')]

        elif(digit == 1 or digit == 5):
            roman = self.RomanDic[str(digit).ljust(place,'0')]
        else:
            if(5 - digit >0):
                roman = self.getRepeatRomanDigit(self.RomanDic['1'.ljust(place,'0')], digit)
            else:
                roman = self.RomanDic['5'.ljust(place,'0')] + self.getRepeatRomanDigit(self.RomanDic['1'.ljust(place,'0')], digit-5)

        return roman

        



    def getRepeatRomanDigit(self, roman:string, repeat:int)->string:
        str = ''
        for x in range(repeat):
            str += roman
        
        return str



    
    def split(self,word):
        return [char for char in word]






intRoman = intRoman(3219)
roman = intRoman.intToRoman() 

print(roman)